src/libraries/oracle/OracleConsulter.sol

```diff
@@ -4,29 +4,46 @@ pragma solidity ^0.8.0;
 import "./OracleLibrary.sol";
 
 library OracleConsulter {
```

Comment:
I chose this design because it was simpler to test and made the contract itself cleaner. Since all the code for consulting the Uni oracle for a TWAP does not require modifying the EVM state, it's reasonable to put all the logic within a library that's linked to the contract on deployment.

As for the method of deployment, wouldn't this be simpler if we utilised Foundry's Solidity Scripting? It seems in that case dynamic linking can be done, which is not possible in the "the more limiting and less user friendly forge create"

Reply:
I believe forge script is good but not what I'm talking about.

Your setup requires deployment of two contracts:

- the external library
- the UniswapV3TWAPOracle contract.